plugins/identity-backend/src/adapters/types.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+
+export type User = {
+  name: string;
+};
+
+export type Group = {
+  name: string;
+  type: string;
+  members?: User[];
+  children?: Group[];
+};
+
+export type GroupsJson = {
+  groups: Group[];
+};
+
+export type GroupsResponse = {
+  groups: Group[];
+};
+
+export interface IdentityApiAdapter {
+  getUserGroups(
```

Comment:
I think it's nicer to move the business logic layer a bit further away from HTTP. So instead of having an interface where you need to implement HTTP req/res logic, you'd model it around request parameters and response body.

Something like

```ts
export type GroupsRequest = {
  user: string
  type: string
}

export type GroupsResponse = {
  groups: Group[];
};

export interface IdentityApi {
  getUserGroups(req: GroupsRequest): Promise<GroupsResponse>
}
```

That way the router layer can be common for all implementations, and you get to do less work in the business logic layer. It'll also be nice to have things decouple from HTTP, if we for example would want to add gRPC or smth down the line. And it also makes it easier to test imo.

Reply:
done!